stock_restrict_lot/models/product_product.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2023 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    # To be able to get the outgoing and incoming qty by the restrict_lot_id
+    # the odoo standard method _compute_quantities_dict is hookable
+    # therefore the domain is already here changed
+    def _get_domain_locations(self):
+        res = super()._get_domain_locations()
+        restrict_lot_id = self.env.context.get("restrict_lot_id")
```

Comment:
It is lot_id in the context according to
https://github.com/odoo/odoo/blob/14.0/addons/stock/models/product.py#L121

```suggestion
        restrict_lot_id = self.env.context.get("lot_id")
```